src/deploy/functions/prepareFunctionsUpload.ts

```diff
@@ -71,8 +78,11 @@ async function packageSource(
   try {
     const files = await fsAsync.readdirRecursive({ path: sourceDir, ignore: ignore });
     for (const file of files) {
+      const name = path.relative(sourceDir, file.name);
+      const fileHash = await getSourceHash(file.name);
+      fileHashes.push(fileHash);
```

Comment:
nit* This seems a bit inefficient given that we have to read the file from disk twice (once to generate a hash, another by the archiver itself.

Can't we create a hash from archived zip once?

Reply:
offline discussion - no b/c metadata changes the hash of the archive...

Reply:
Unfortunately we can't hash the archived zip due to metadata. 🐼